src/hotspot/share/classfile/systemDictionaryShared.cpp

```diff
@@ -995,6 +1022,24 @@ void SystemDictionaryShared::check_verification_constraints(InstanceKlass* klass
   }
 }
 
+void SystemDictionaryShared::copy_verification_constraints_from_preimage(InstanceKlass* klass) {
+  assert(CDSConfig::is_using_archive(), "called at run time with CDS enabled only");
+  DumpTimeClassInfo* dt_info = get_info(klass);
+  RunTimeClassInfo* rt_info = RunTimeClassInfo::get_for(klass); // from preimage
+
+  int length = rt_info->num_verifier_constraints();
+  if (length > 0 || klass->name()->equals("HelloWorld")) {
```

Comment:
Is the "HelloWorld" check leftover from debugging?

Reply:
Fixed.